diff -uNr rootfs.original/home/lava/klipper/klippy/extras/filament_detect.py rootfs/home/lava/klipper/klippy/extras/filament_detect.py
--- rootfs.original/home/lava/klipper/klippy/extras/filament_detect.py
+++ rootfs/home/lava/klipper/klippy/extras/filament_detect.py
@@ -1,5 +1,6 @@
 import logging, copy, os
 from . import filament_protocol
+from . import filament_protocol_openspool
 from . import fm175xx_reader
 from . import filament_feed

@@ -118,12 +119,20 @@
         if (fm175xx_reader.FM175XX_CARD_INFO_READ == operation):
             if (fm175xx_reader.FM175XX_MIFARE_CARD_TYPE_M1 == card_type and  fm175xx_reader.FM175XX_OK == result):
                 logging.info("channel[%d] m1 card data parsing....", channel)
-                error, info = filament_protocol.m1_proto_data_parse(card_data)
+                error, info = filament_protocol_openspool.m1_proto_data_parse(card_data)
                 if (error == filament_protocol.FILAMENT_PROTO_OK):
                     logging.info("channel[%d] m1 parse ok....", channel)
                     filament_info = info
                 else:
                     logging.error("channel[%d] m1 parse err: %d", channel, error)
+            elif fm175xx_reader.FM175XX_MIFARE_CARD_TYPE_NTAG == card_type and fm175xx_reader.FM175XX_OK == result:
+                logging.info("channel[%d] trying NDEF parsing....", channel)
+                error, info = filament_protocol_openspool.ndef_proto_data_parse(card_data)
+                if (error == filament_protocol.FILAMENT_PROTO_OK):
+                    logging.info("channel[%d] NDEF parse ok....", channel)
+                    filament_info = info
+                else:
+                    logging.error("channel[%d] NDEF parse err....", channel)
         else:
             is_clear = True
