--- a/printer.cfg
+++ b/printer.cfg
@@ -54,29 +54,53 @@
 #################################################################
 ###       NFC configuration
 #################################################################
-[fm175xx_reader]
-soc_spi_bus: 2
-soc_spi_dev_num: 0
-soc_spi_mode: 0
-soc_spi_speed_max: 500000
-soc_rst_pin: host:gpiochip1/gpio25
-
-extra_spi_bus: 2
-extra_spi_dev_num: 1
-extra_spi_mode: 0
-extra_spi_speed_max: 500000
-extra_rst_pin: host:gpiochip1/gpio28
+#[fm175xx_reader]
+#soc_spi_bus: 2
+#soc_spi_dev_num: 0
+#soc_spi_mode: 0
+#soc_spi_speed_max: 500000
+#soc_rst_pin: host:gpiochip1/gpio25
+
+#extra_spi_bus: 2
+#extra_spi_dev_num: 1
+#extra_spi_mode: 0
+#extra_spi_speed_max: 500000
+#extra_rst_pin: host:gpiochip1/gpio28
 
 # Equivalently: etruder ~ extruder3
-soc_ch_1: 3
-soc_ch_2: 2
-extra_ch_1: 0
-extra_ch_2: 1
+#soc_ch_1: 3
+#soc_ch_2: 2
+#extra_ch_1: 0
+#extra_ch_2: 1
 
-rf_1_pin: host:gpiochip1/gpio27
-rf_2_pin: host:gpiochip1/gpio24
+#rf_1_pin: host:gpiochip1/gpio27
+#rf_2_pin: host:gpiochip1/gpio24
 
-[filament_detect]
+#[filament_detect]
+
+[gcode_macro FILAMENT_DT_UPDATE]
+gcode:
+    {% set channel = params.CHANNEL|default(0)|int %}
+    { action_call_remote_method("filament_detect.start_scan", slot=channel) }
+
+[gcode_macro FILAMENT_DT_CLEAR]
+gcode:
+    {% set channel = params.CHANNEL|default(0)|int %}
+    SET_PRINT_FILAMENT_CONFIG CONFIG_EXTRUDER={channel} VENDOR=NONE FILAMENT_TYPE=NONE FILAMENT_SUBTYPE=NONE FILAMENT_COLOR=0
+
+[gcode_macro FILAMENT_DT_QUERY]
+gcode:
+    {% set channel = params.CHANNEL|default(0)|int %}
+    {% set cfg = printer.print_task_config %}
+    {% if channel < 0 or channel >= cfg.filament_vendor | length %}
+        { action_raise_error("channel[%d] is out of range[0, %d]" % (channel, (cfg.filament_vendor | length) - 1)) }
+    {% endif %}
+    { action_respond_info("channel[%d] vendor = %s, main_type: %s, sub_type= %s, rgba_color = %08X"
+        % (channel,
+           cfg.filament_vendor[channel],
+           cfg.filament_type[channel],
+           cfg.filament_sub_type[channel],
+           cfg.filament_color[channel])) }
 
 #################################################################
 ###       flow calibration configuration
