--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,7 +8,14 @@
 
 # LIBVNC
 find_library(LIBVNC NAMES libvncserver vncserver)
-target_link_libraries(framebuffer-vncserver ${LIBVNC})
+target_link_libraries(framebuffer-vncserver
+  ${LIBVNC}
+  /usr/lib/aarch64-linux-gnu/libz.a
+  /usr/lib/aarch64-linux-gnu/liblzo2.a
+  /usr/lib/aarch64-linux-gnu/libgcrypt.a
+  /usr/lib/aarch64-linux-gnu/libgpg-error.a
+  /usr/lib/aarch64-linux-gnu/libsasl2.a
+  /usr/lib/aarch64-linux-gnu/libcrypto.a)
 
 
 MESSAGE( STATUS "LIBVNC:         " ${LIBVNC} )
