--- a/klippy/extras/filament_feed.py
+++ b/klippy/extras/filament_feed.py
@@ -187,8 +187,10 @@
             self.white_light.set_pwm(print_time, 0, FEED_MIN_TIME)
 
 class FeedPort:
-    def __init__(self, printer, reactor, pin, threshold):
+    def __init__(self, printer, reactor, pin, threshold, index):
         self.reactor = reactor
+        self.index = index
+        self.ace = None
         ppins = printer.lookup_object('pins')
         self._port = ppins.setup_pin('adc', pin)
         self._port_adc_value = 0
@@ -226,7 +228,12 @@
     def get_adc_value(self):
         return self._port_adc_value
 
+    def add_ace(self, ace):
+        self.ace = ace
+
     def get_filament_detected(self):
+        if self.ace is not None:
+            return self.ace.get_filament_detect(self.index)
         return self._filament_detected
 
 class FeedTachometer:
@@ -366,6 +373,7 @@
         self.reactor = self.printer.get_reactor()
         self.gcode = self.printer.lookup_object('gcode')
         self.module_name = config.get_name().split()[1]
+        self.ace = None
 
         self.channel_active = None
         self.channel_state = [FEED_STA_NONE] * FEED_CHANNEL_NUMS
@@ -412,12 +420,12 @@
         self._port = []
         tmp_pin = config.get('port_ch_1_pin')
         threshold = config.getfloat('port_ch_1_threshold')
-        tmp_obj = FeedPort(self.printer, self.reactor, tmp_pin, threshold)
+        tmp_obj = FeedPort(self.printer, self.reactor, tmp_pin, threshold, self.filament_ch[0])
         tmp_obj.register_cb_2_port_event(self._port_ch1_event_handler)
         self._port.append(tmp_obj)
         tmp_pin = config.get('port_ch_2_pin')
         threshold = config.getfloat('port_ch_2_threshold')
-        tmp_obj = FeedPort(self.printer, self.reactor, tmp_pin, threshold)
+        tmp_obj = FeedPort(self.printer, self.reactor, tmp_pin, threshold, self.filament_ch[1])
         tmp_obj.register_cb_2_port_event(self._port_ch2_event_handler)
         self._port.append(tmp_obj)
         self.gcode.register_mux_command("FEED_PORT", "MODULE",
@@ -542,6 +550,9 @@
     def _ready(self):
         self.toolhead = self.printer.lookup_object('toolhead')
         self.gcode_move = self.printer.lookup_object('gcode_move')
+        self.ace = self.printer.lookup_object('ace', None)
+        for i in self._port:
+            i.add_ace(self.ace)
         self.exception_manager = self.printer.lookup_object('exception_manager', None)
         self.reactor.update_timer(self._check_init_state_timer,
                                   self.reactor.monotonic() + 2 * FEED_PORT_ADC_REPORT_TIME)
@@ -562,7 +573,7 @@
         self.reactor.unregister_timer(self._check_init_state_timer)
 
         for ch in range(FEED_CHANNEL_NUMS):
-            if self._port[ch].get_adc_value() < FEED_PORT_ADC_VAL_MODULE_EXIST:
+            if self._port[ch].get_adc_value() < FEED_PORT_ADC_VAL_MODULE_EXIST or self.ace is not None:
                 self.module_exist[ch] = True
             else:
                 self.module_exist[ch] = False
@@ -807,5 +818,7 @@
             # preload
             elif action == FEED_ACT_PRELOAD:
+                if self.ace is not None:
+                    return
                 has_put_into_drive = False
                 try:
                     self.exception_code[ch] = 10
@@ -821,7 +832,8 @@
                     # feed
                     self._set_channel_state(ch, FEED_STA_PRELOAD_FEEDING)
                     systime_1 = self.reactor.monotonic()
-                    self.motor.run(motor_dir, self.motor_speed_slow_switching)
+                    if self.ace is None:
+                        self.motor.run(motor_dir, self.motor_speed_slow_switching)
                     has_put_into_drive = True
                     arrive_runout_sensor = False
                     logging.info("[feed] extruder[%d] putting into drive", self.filament_ch[ch])
@@ -830,14 +842,17 @@
                     preload_duty = self.motor_speed_preload
                     if arrive_runout_sensor == False:
                         for i in range(3):
-                            self.motor.run(motor_dir, preload_duty)
+                            if self.ace is None:
+                                self.motor.run(motor_dir, preload_duty)
                             self.reactor.pause(self.reactor.monotonic() + 0.35)
                             arrive_runout_sensor = self.runout_sensor[ch].get_status(0)['filament_detected']
                             if arrive_runout_sensor == True or preload_duty >= 1.0:
                                 break
-                            preload_duty = min(1.0, preload_duty + 0.1)
+                            if self.ace is None:
+                                preload_duty = min(1.0, preload_duty + 0.1)
                         if arrive_runout_sensor == False and preload_duty < 1.0:
-                            self.motor.run(motor_dir, 1.0)
+                            if self.ace is None:
+                                self.motor.run(motor_dir, 1.0)
                             self.reactor.pause(self.reactor.monotonic() + 0.2)
                             arrive_runout_sensor = self.runout_sensor[ch].get_status(0)['filament_detected']
                     logging.info("[feed_preload] extruder[%d], duty:%f, ", self.filament_ch[ch], preload_duty)
@@ -996,6 +1011,28 @@
                         one_step_cnt = self.wheel[ch].ppr * 2.0 * 10.0 / FEED_WHEEL_CIRCUMFERENCE
 
+                        if self.ace is not None:
+                            self.ace.start_load_feed(self.filament_ch[ch])
                         while 1:
+                            if self.ace is not None:
+                                self.reactor.pause(self.reactor.monotonic() + 0.105)
+                                systime_2 = self.reactor.monotonic()
+                                port_detect = self._port[ch].get_filament_detected()
+                                runout_detect = self.runout_sensor[ch].get_status(0)['filament_detected']
+                                if runout_detect == True:
+                                    self.ace.stop_load_feed(self.filament_ch[ch], "sensor_hit")
+                                    self.channel_error[ch] = FEED_OK
+                                    break
+                                if port_detect == False:
+                                    self.ace.stop_load_feed(self.filament_ch[ch], "no_filament")
+                                    self.channel_error[ch] = FEED_ERR_NO_FILAMENT
+                                    self.exception_code[ch] = 33
+                                    break
+                                if systime_2 - systime_0 > FEED_LOAD_TIMEOUT_TIME:
+                                    self.ace.stop_load_feed(self.filament_ch[ch], "timeout")
+                                    self.channel_error[ch] = FEED_ERR_TIMEOUT
+                                    self.exception_code[ch] = 34
+                                    break
+                                continue
                             wheel_cnt_a_1 = self.wheel[ch].get_counts()
                             wheel_cnt_b_1 = self.wheel_2[ch].get_counts()
                             motor_cnt_1 = self.motor_tachometer.get_counts()
